### PR TITLE
Delete Dead Code from DI

### PR DESCRIPTION
**Pull request by gottesmm**
Just deleting dead code. DefiniteInitialization has the property that MemoryInst is always a MarkUninitializedInst. So anything that will never run if MemoryInst is a MarkUninitializedInst is trivially dead.

### CODE REVIEW

**Comment by gottesmm** [comment]
This is the dead code removal moved out of #16844.

**Comment by gottesmm** [comment]
@swift-ci smoke test and merge

File: lib/SILOptimizer/Mandatory/DIMemoryUseCollectorOwnership.cpp [hunk 1]
```diff
@@ -62,19 +62,9 @@ static unsigned getElementCountRec(SILModule &Module, SILType T,
...
-           "analyzing multi-field boxes not implemented");
-    return {ABI->getBoxType()->getFieldType(MemoryInst->getModule(), 0), false};
-  }
-
-  if (auto *ASI = dyn_cast<AllocStackInst>(MemoryInst)) {
-    return {ASI->getElementType(), false};
-  }
-
-  auto *MUI = cast<MarkUninitializedInst>(MemoryInst);
+  auto *MUI = MemoryInst;
```
Comment (slavapestov) [inline lib/SILOptimizer/Mandatory/DIMemoryUseCollectorOwnership.cpp]: Remove MUI since you never re-assign it?

Comment (gottesmm) [reply lib/SILOptimizer/Mandatory/DIMemoryUseCollectorOwnership.cpp]: Yes.

**Pull request merged**